console/services/market_app/new_components.py

```diff
@@ -256,6 +258,36 @@ def _template_to_ports(self, component, ports):
                 gateway_rules.append(gateway_rule)
         return new_ports, gateway_rules
 
+    def _ensure_port_envs(self, component, envs, ports, governance_mode):
+        # filter out the old port envs
+        envs = [env for env in envs if env.container_port == 0]
+        # create outer envs for every port
+        for port in ports:
+            envs.extend(self._create_envs_4_ports(component, port, governance_mode))
+        return envs
+
+    def _create_envs_4_ports(self, component: TenantServiceInfo,  port: TenantServicesPort, governance_mode):
```

Comment:
If it already exists, will it be created again?

Reply:
No. It will delete the old ones, then create the new ones.